### ISSUE TITLE
parallel execution in validation phase

### ISSUE THREAD

**Issue by kd1yadav** [sysrepo/sysrepo#2500]
` edit-config ` request is processed by sysrepo in two phase.
First phase is  validation (invoke callback with ` sr_ev_verify `).
Second  phase is apply or commit (invoke callback with ` sr_ev_apply `) - in this phase lock is required.
Other thing is netopeer2 sessions are independent to each other and their RPCs execute in parallel.
From above observations can I reach to this conclusion that two edit-config(targeting different yang data models) from two netopeer2 session execute parallel in validation phase?

**Comment by michalvasko** [comment]
NETCONF sessions are independent, generally speaking, but specific RPC (`edit-config`) can always execute only one at a time. So, edit-config will never be executed in parallel. But your observations are correct and edits performed locally in *sysrepo* (such as using `sysrepocfg`) *may* be executed in parallel (in case the data being edited do not depend on each other).

**Comment by kd1yadav** [comment]
Thanks @michalvasko

**Comment by kd1yadav** [comment]
Thanks @michalvasko for your prompt reply.
My question was that since `sr_ev_verify` does not take any locking can we somehow introduce concurrency in it through client and not through `sysrepocfg`?

I would like to explain my scenario:
We have an use case where we are using 2 devices embedded on a single card and trying to manage these 2 devices. There is a single `sysrepo-datastore` and `netopeer2-server` on that card. 
Can we manage these 2 device's data in parallel as the data is completely different ( both devices use completely different yang models)? Can it be done to any extent if not fully?

**Comment by michalvasko** [comment]
It seems this is a duplicate of #2495, I have answered there. But seeing as this is a requested feature I will think about possible solutions.
